server/weaverbird/steps/duration.py

```diff
@@ -0,0 +1,36 @@
+from pandas import DataFrame
+from pydantic import Field
+
+from weaverbird.steps import BaseStep
+from weaverbird.types import ColumnName, DomainRetriever, PipelineExecutor
+
+_SECOND = 1
+_MINUTE = _SECOND * 60
+_HOUR = _MINUTE * 60
+_DAY = _HOUR * 24
+
+durations_in_second = {'seconds': _SECOND, 'minutes': _MINUTE, 'hours': _HOUR, 'days': _DAY}
+
+
+class DurationStep(BaseStep):
+    name: str = Field('duration', const=True)
+    new_column_name: ColumnName = Field(alias='newColumnName')
+    start_date_column: ColumnName = Field(alias='startDateColumn')
+    end_date_column: ColumnName = Field(alias='endDateColumn')
+    duration_in: ColumnName = Field(alias='durationIn')
```

Comment:
```suggestion
    duration_in: Literal['seconds', 'minutes', 'hours', 'days'] = Field(alias='durationIn')
```
(et enlever le `assert`)